src/App.js

```diff
@@ -1,10 +1,51 @@
+import React, { useEffect, useState, useCallback } from 'react';
+
+import GlobalStyle from './styles/GlobalStyle';
+import { Main, Section } from './styles/Container';
+import { getLocalStorage, syncLocalStorage } from './components/TodoState';
+import { runToast, ToastComponent } from './components/toast/Toast';
+import InputTodo from './components/InputTodo';
+import TodoItemList from './components/TodoItemList';
 
 function App() {
-  return (
-    <div>
-      <h1>15기 화이팅~</h1>
-    </div>
-  );
+    const [list, setList] = useState(getLocalStorage);
+
+    // list 상태 변화할 때마다 로컬스토리지 동기화
+    useEffect(() => syncLocalStorage(list), [list]);
+
+    const toggleTodo = useCallback(
+        (id) => {
+            setList((list) => list.map((item) => (item.id === id ? { ...item, done: !item.done } : item)));
+        },
+        [list]
+    );
+
+    const removeTodo = useCallback(
+        (id) => {
+            setList((list) => list.filter((item) => item.id !== id));
+            runToast('삭제되었어요!');
+        },
+        [list]
+    );
```

Comment:
state를 변경하실 때 함수형 업데이트를 사용하셔서 `useCallback`의 dependency array에 `list`는 불필요해 보입니다!

```suggestion
    const toggleTodo = useCallback(
        (id) => {
            setList((list) => list.map((item) => (item.id === id ? { ...item, done: !item.done } : item)));
        },
        []
    );

    const removeTodo = useCallback(
        (id) => {
            setList((list) => list.filter((item) => item.id !== id));
            runToast('삭제되었어요!');
        },
        []
    );
```

Reply:
감사합니다. 좀더 공부를 해봐야겠네요